src/Led.cpp

```diff
@@ -224,500 +259,587 @@ static void Led_Task(void *parameter) {
+							}
 						} else {
-							leds[Led_Address(i) % NUM_LEDS] = CRGB::BlueViolet;
-							leds[(Led_Address(i) + NUM_LEDS / 4) % NUM_LEDS] = CRGB::BlueViolet;
-							leds[(Led_Address(i) + NUM_LEDS / 2) % NUM_LEDS] = CRGB::BlueViolet;
-							leds[(Led_Address(i) + NUM_LEDS / 4 * 3) % NUM_LEDS] = CRGB::BlueViolet;
-						}
-						FastLED.show();
-						if (gPlayProperties.playMode != BUSY) {
-							break;
+							generalColor = CRGB::Orange;
+							if (OPMODE_BLUETOOTH_SOURCE == System_GetOperationMode()) {
+								generalColor = CRGB::Blue;
+							} else
+							if (gPlayProperties.currentSpeechActive) {
+								generalColor = speechColor;
+							}
+
+							uint8_t pauseOffset = 0;
+							if (OFFSET_PAUSE_LEDS) {
+								pauseOffset = ((NUM_LEDS/NUM_LEDS_IDLE_DOTS)/2)-1;
+							}
+							Led_DrawIdleDots(leds, pauseOffset, generalColor);
 						}
-						vTaskDelay(portTICK_RATE_MS * 50);
-					}
-				}
-				break;
-
-			default: // If playlist is active (doesn't matter which type)
-				if (!gPlayProperties.playlistFinished) {
-					if (Led_NeedsProgressRedraw(lastPlayState, lastLockState,
-								    requestProgressRedraw)) {
-						lastPlayState = gPlayProperties.pausePlay;
-						lastLockState = System_AreControlsLocked();
-						requestProgressRedraw = false;
-						redrawProgress = true;
-					}
-
-					if (requestClearLeds) {
-						FastLED.clear();
 						FastLED.show();
-						requestClearLeds = false;
-					}
-
-					// Single-LED: led indicates between gradient green (beginning) => red (end)
-					// Multiple-LED: growing number of leds indicate between gradient green (beginning) => red (end)
-					if (!gPlayProperties.isWebstream) {
-						if (gPlayProperties.currentRelPos != lastPos || redrawProgress) {
-							redrawProgress = false;
+						animaitonTimer = 10;
+					} break;
+
+					case LedAnimationType::Progress: {
+						if (gPlayProperties.currentRelPos != lastPos || animationIndex == 0) {
+							animationIndex = 1;
 							lastPos = gPlayProperties.currentRelPos;
 							FastLED.clear();
 							if (NUM_LEDS == 1) {
 								leds[0].setHue((uint8_t)(85 - ((double)90 / 100) * gPlayProperties.currentRelPos));
 							} else {
-								uint8_t numLedsToLight = map(gPlayProperties.currentRelPos, 0, 98, 0, NUM_LEDS);
-								for (uint8_t led = 0; led < numLedsToLight; led++) {
+								uint32_t ledValue = map(gPlayProperties.currentRelPos, 0, 98, 0, NUM_LEDS * DIMMABLE_STATES);
+								uint8_t fullLeds = ledValue/DIMMABLE_STATES;
+								uint8_t lastLed = ledValue%DIMMABLE_STATES;
+								for (uint8_t led = 0; led < fullLeds; led++) {
 									if (System_AreControlsLocked()) {
 										leds[Led_Address(led)] = CRGB::Red;
 									} else if (!gPlayProperties.pausePlay) { // Hue-rainbow
 										leds[Led_Address(led)].setHue((uint8_t)(((float)PROGRESS_HUE_END - (float)PROGRESS_HUE_START) / (NUM_LEDS-1) * led + PROGRESS_HUE_START));
 									}
 								}
```

Comment:
Didn't look cleaner when I tried this in my opinion.
And since we also want the last "dimed" LED in the right color we would have to evalute first if this LED is greater 0 and after all LEDs (including the dimed) are in the right color modify the last LED again.
I would suggest to leave it like that for the moment.

Reply:
Yes, I see what you mean. Either we move the `if(lastLed > 0)` into the controlls-locked if and duplicate it, or we leave it as it is now.

Also we are accessing the leds in the loop with Led_Address, which takes offset & direction into account. I did not realised that yesterday.

fill_solid does not, so we can not use it anywhere, where we are not filling up all LEDs. Otherwise the LEDs would be in the wrong offset & direction. As we say in Austria: "Das wäre gar nicht gut" (like, "that's would not be good" meaning, "that'd be **really** bad")

That's a real bummer. 


(thinking out loud)
What we basically would need is a offset & direction aware RingBuffer to replace the needed calls to Led_Address (something like an extended CRGBSet, which is already direction aware). But that would be a bigger rework, propably a whole new PR based on this one.

Reply:
All fine. Thanks anyways! If you want we can also discuss this in german in the forum a little. There already is a topic to this Pull-Request.